src/main/java/uk/gov/hmcts/reform/cwrdapi/repository/IdamRepository.java

```diff
@@ -17,6 +18,7 @@ public IdamRepository(IdamClient idamClient) {
         this.idamClient = idamClient;
     }
 
+    @Cacheable(value = "token")
```

Comment:
you can add logger to check how many times the call is happening.

Reply:
Addressed here: https://github.com/hmcts/rd-caseworker-ref-api/pull/121/commits/d35f0f9330baaa11eb91411b61c343c8d763343e